OpenRA.Mods.Common/TargetExtensions.cs

```diff
@@ -31,20 +31,20 @@ public static Target RecalculateInvalidatingHiddenTargets(this Target t, Player
 		/// If the target actor becomes hidden without a FrozenActor, the target actor is kept
 		/// and the actorHidden flag is set to true.
 		/// </summary>
-		public static Target Recalculate(this Target t, Player viewer, out bool targetIsHiddenActor)
+		public static Target Recalculate(this Target t, Player viewer, out bool targetIsHiddenActor, bool allowOwnerChange = false)
 		{
 			targetIsHiddenActor = false;
 
 			// Check whether the target has transformed into something else
 			// HACK: This relies on knowing the internal implementation details of Target
-			if (t.Type == TargetType.Invalid && t.Actor != null && t.Actor.ReplacedByActor != null)
+			if (t.GetTargetType(allowOwnerChange) == TargetType.Invalid && t.Actor != null && t.Actor.ReplacedByActor != null)
```

Comment:
IMO pull `t.GetTargetType(allowOwnerChange)` into a local for the sake of readability.